### ISSUE TITLE
pyparsing 3.1.3 fails on Python 3.6

### ISSUE THREAD

**Issue by felixfontein** [pyparsing/pyparsing#564]
The class `re.Pattern` only exists since Python 3.7, but is used in https://github.com/pyparsing/pyparsing/blob/master/pyparsing/core.py#L3060.

```
  Traceback (most recent call last):
    File "/home/runner/work/antsibull-docs-parser/antsibull-docs-parser/antsibull-docs-parser/.nox/test-3-6/lib/python3.6/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 363, in <module>
      main()
    File "/home/runner/work/antsibull-docs-parser/antsibull-docs-parser/antsibull-docs-parser/.nox/test-3-6/lib/python3.6/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 345, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "/home/runner/work/antsibull-docs-parser/antsibull-docs-parser/antsibull-docs-parser/.nox/test-3-6/lib/python3.6/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 160, in prepare_metadata_for_build_wheel
      whl_basename = backend.build_wheel(metadata_directory, config_settings)
    File "/tmp/pip-build-env-1qhbq5lc/overlay/lib/python3.6/site-packages/hatchling/build.py", line 41, in build_wheel
      return os.path.basename(next(builder.build(wheel_directory, ['standard'])))
    File "/tmp/pip-build-env-1qhbq5lc/overlay/lib/python3.6/site-packages/hatchling/builders/plugin/interface.py", line 76, in build
      self.metadata.core.validate_fields()
    File "/tmp/pip-build-env-1qhbq5lc/overlay/lib/python3.6/site-packages/hatchling/metadata/core.py", line 1115, in validate_fields
      getattr(self, attribute)
    File "/tmp/pip-build-env-1qhbq5lc/overlay/lib/python3.6/site-packages/hatchling/metadata/core.py", line 1014, in dependencies
      self._dependencies = sorted(self.dependencies_complex)
    File "/tmp/pip-build-env-1qhbq5lc/overlay/lib/python3.6/site-packages/hatchling/metadata/core.py", line 971, in dependencies_complex
      from packaging.requirements import InvalidRequirement, Requirement
    File "/tmp/pip-build-env-1qhbq5lc/overlay/lib/python3.6/site-packages/packaging/requirements.py", line 10, in <module>
      from pyparsing import (  # noqa
    File "/tmp/pip-build-env-1qhbq5lc/overlay/lib/python3.6/site-packages/pyparsing/__init__.py", line 132, in <module>
      from .core import __diag__, __compat__
    File "/tmp/pip-build-env-1qhbq5lc/overlay/lib/python3.6/site-packages/pyparsing/core.py", line 2988, in <module>
      class Regex(Token):
    File "/tmp/pip-build-env-1qhbq5lc/overlay/lib/python3.6/site-packages/pyparsing/core.py", line 3060, in Regex
      def re(self) -> re.Pattern:
  AttributeError: module 're' has no attribute 'Pattern'
```
(Ref: https://github.com/ansible-community/antsibull-docs-parser/actions/runs/10545127361/job/29214931239)

**Comment by ngie-eign** [comment]
Could y'all please pull the prior release so it doesn't get automatically installed as part of dependent packages, like tox?

**Comment by felixfontein** [comment]
3.1.4 just got released (< 10 minutes ago) which fixes this problem.

**Comment by ptmcg** [comment]
@ngie-eign I've pushed version 3.1.4 that resolves the Python 3.6 incompatibility. Is that sufficient?

**Comment by ngie-eign** [comment]
That works -- thank you folks!!
